docs/content/index.md

```diff
@@ -2,60 +2,153 @@
+Individually, the events are not very interesting, however, summarizations of this type of data can yield many useful insights.
+Druid can optionally choose to summarize this raw data at ingestion time using a process we refer to as "roll-up".
+Roll-up is a first-level aggregation operation over a selected set of dimensions, equivalent to (in pseudocode):
 
+    GROUP BY timestamp, publisher, advertiser, gender, country
+      :: impressions = COUNT(1),  clicks = SUM(click),  revenue = SUM(price)
 
-When Druid?
-----------
+The compacted version of our original raw data looks something like this:
 
-* You need to do interactive aggregations and fast exploration on large amounts of data
-* You need ad-hoc analytic queries (not a key-value store)
-* You have a lot of data (10s of billions of events added per day, 10s of TB of data added per day)
-* You want to do your analysis on data as it’s happening (in real-time)
-* You need a data store that is always available with no single point of failure.
+     timestamp             publisher          advertiser  gender country impressions clicks revenue
+     2011-01-01T01:00:00Z  ultratrimfast.com  google.com  Male   USA     1800        25     15.70
+     2011-01-01T01:00:00Z  bieberfever.com    google.com  Male   USA     2912        42     29.18
+     2011-01-01T02:00:00Z  ultratrimfast.com  google.com  Male   UK      1953        17     17.31
+     2011-01-01T02:00:00Z  bieberfever.com    google.com  Male   UK      3194        170    34.01
 
-Architecture Overview
----------------------
+In practice, we see that rolling up data can dramatically reduce the size of data that needs to be stored (often by a factor of 100).
+This storage reduction does come at a cost; as we roll up data, we lose the ability to query individual events. Phrased another way,
+the rollup granularity is the minimum granularity you will be able to query data at. Hence, Druid ingestion specs define this granularity as the `queryGranularity` of the data.
+
+Druid is designed to perform single table operations and does not currently support joins.
+Although many production setups instrument joins at the ETL level, data must be denormalized before it is loaded into Druid.
+
+## Indexing the Data
+
+Druid gets its speed in part from how it stores data. Borrowing ideas from search infrastructure,
+Druid creates immutable snapshots of data, stored in data structures highly optimized for analytic queries.
+
+Druid is a column store, which means each individual column is stored separately. Only the columns that pertain to a query are used
+in that query, and Druid is pretty good about only scanning exactly what it needs for a query.
+Different columns can also employ different compression methods. Different columns can also have different indexes associated with them.
+
+## Sharding the Data
```

Comment:
Consider putting sharding before indexing to stress the fact that indexing is per-shard.